4_object_factory/node_4.x/addon.cc

```diff
@@ -0,0 +1,18 @@
+#include <node.h>
+
+using namespace v8;
+
+void CreateObject(const FunctionCallbackInfo<Value>& args) {
+  Isolate* isolate = args.GetIsolate();
+
+  Local<Object> obj = Object::New(isolate);
+  obj->Set(String::NewFromUtf8(isolate, "msg"), args[0]->ToString());
```

Comment:
It's perhaps overwrought for an example but `ToString()` can fail and return an empty handle, e.g. when you call it with `addon({ toString() { throw 'boom' } })`.